src/gsi/m_berror_stats.f90

```diff
@@ -125,6 +131,52 @@ subroutine get_dims(msig,mlat,mlon,lunit)
    return
 end subroutine get_dims
 
+subroutine inquire_berror(lunit,mype)
+
+   use kinds,    only: r_single
+
+   implicit none
+
+   integer(i_kind),intent(in   ) :: lunit ! logical unit [22]
+   integer(i_kind),intent(in   ) :: mype
+
+   character(len=*),parameter :: myname_=myname//'::inquire_berror'
+   character(len=5) :: var
+   integer(i_kind) :: inerr,msig,mlat,mlon_,ier,isig,errtot,i
+   real(r_single),dimension(:),allocatable::  clat_avn,sigma_avn
+
+   ! Read dimension of stats file
+   inerr = lunit
+   open(inerr,file=berror_stats,form='unformatted',status='old',iostat=ier)
+   call check_iostat(ier,myname_,'open('//trim(berror_stats)//')')
+   rewind inerr
+   read(inerr,iostat=ier) msig,mlat,mlon_
+   call check_iostat(ier,myname_,'read header')
+   errtot=ier
+
+   errtot=0
+   allocate ( clat_avn(mlat) )
+   allocate ( sigma_avn(1:msig) )
+   read(inerr,iostat=ier)clat_avn,sigma_avn
+   do i=1,msig-1
+     if(sigma_avn(i) < sigma_avn(i+1))then
```

Comment:
can some explanation be added here for why such a test could  determine the logical value of usenewgfsberror?

Reply:
Ting,

This test is just to see if the second record contains the expected values for the sigma_avn array.  It is possible, but very unlikely that if the wrong record is read, this criterion would be satisfied for all levels.